frag-nav/src/main/java/com/ncapdevi/fragnav/FragNavController.java

```diff
@@ -589,6 +621,37 @@ private void clearFragmentManager() {
             executePendingTransactions();
         }
     }
+
+    /**
+     * Setup a fragment transaction with the given option
+     *
+     * @param transactionOptions The options that will be set for this transaction
+     */
+    @CheckResult
+    private FragmentTransaction createTransactionWithOptions(@Nullable FragNavTransactionOptions transactionOptions) {
+        FragmentTransaction ft = mFragmentManager.beginTransaction();
+        if (transactionOptions != null) {
+            if (transactionOptions.transition != null) {
+                ft.setTransition(transactionOptions.transition);
+            } else {
+                ft.setTransition(mDefaultTransitionMode);
+            }
+
+            if (transactionOptions.sharedElements != null) {
+                for (Pair<View, String> sharedElement : transactionOptions.sharedElements) {
+                    ft.addSharedElement(sharedElement.first, sharedElement.second);
+                }
+            }
+            if (transactionOptions.enterAnimationId != null) {
```

Comment:
This will fail to add `exitAnimation` if `enterAnimation` is null. Since `FragmentTransaction` supports setting these to, `null|0` you might as well always call `setCustomAnimations`.